File: c++ DroneSim/droneplus.cpp
#include "droneplus.h"
#include "eccezione.h"
#include <cmath>
using std::abs;

dronePlus::dronePlus(bool o, int b, int x, int y):drone(o,b,x,y){}

string dronePlus::getName()const{
    return "Drone Plus";
}

coord dronePlus::move(int x, int y){
    if( (x==0 && abs(y)==1) || (y==0 && abs(x)==1) ){
        return drone::move(x,y);
    }
    else{
        throw eccezione('m');
    }

}

int dronePlus::batteryUsage = 3;

File: java DroneSim/DronePlus.java
class DronePlus extends Drone{
	private static int batteryUsage = 3;

    public DronePlus(boolean s, int b, int x, int y){
    	super(s,b,x,y);
    }
    public DronePlus(){
		this(false, 100, 0, 0);
	}


    public String getName(){
    	return "Drone Plus";
    }

    public Coord move(int x, int y) throws Eccezione {
    	if( (x==0 && Math.abs(y)==1) || (y==0 && Math.abs(x)==1) ){
	        return super.move(x,y);
	    }
	    else{
	        throw new Eccezione('m');
	    }
    }

    public Coord move() throws Eccezione {
    	return this.move(0,1);
    }
}